src/platform/nrfconnect/OTAImageProcessorImpl.cpp

```diff
@@ -114,6 +144,10 @@ CHIP_ERROR OTAImageProcessorImpl::Apply()
     // Schedule update of all images
     int err = dfu_target_schedule_update(-1);
 
+#ifdef CONFIG_THREAD_WIFI_SWITCHING
+    err = OTAMultiImageDownloaders::Apply();
```

Comment:
Shouldn't it be rather called in `ConfirmCurrentImage`? That is, when we know that the image has been applied correctly and we already execute it successfully?

Reply:
Ahh sure, my bad.